src/frontend/src/optimizer/plan_node/logical_source.rs

```diff
@@ -85,11 +85,13 @@ impl LogicalSource {
         let base = PlanBase::new_logical_with_core(&core);
 
         let output_exprs = Self::derive_output_exprs_from_generated_columns(&core.column_catalog)?;
+        let (core, output_row_id_index) = core.exclude_generated_columns();
```

Comment:
I move the logic in `rewrite_to_stream_batch_source` here, which can help simplify the logic in `to_stream` a lot.